zebra-network/src/protocol/internal/request.rs

```diff
@@ -39,11 +42,30 @@ pub enum Request {
     /// didn't start with a `Vec` but with, e.g., an iterator, they can collect
     /// directly into a `HashSet` and save work.
     ///
+    /// If this requests a recently-advertised block, the peer set will make a
+    /// best-effort attempt to route the request to a peer that advertised the
+    /// block. In this case, it's best to use a request set of size 1.
+    /// Otherwise, it is routed using the normal load-balancing strategy.
+    ///
     /// # Returns
     ///
     /// Returns [`Response::Blocks`](super::Response::Blocks).
     BlocksByHash(HashSet<block::Hash>),
 
+    /// Request transactions by hash.
+    ///
+    /// This uses a `HashSet` for the same reason as [`Request::BlocksByHash`].
+    ///
+    /// If this requests a recently-advertised transaction, the peer set will
+    /// make a best-effort attempt to route the request to a peer that advertised
+    /// the transaction. In this case, it's best to use a request set of size 1.
```

Comment:
Similarly, isn't the size a requirement?
```suggestion
    /// the transaction. This routing is only used for request sets of size 1.
```